<launch>

  <arg name="model" default="$(find wpv3_bringup)/urdf/wpv3.urdf"/>
  <arg name="gui" default="true" />
  <arg name="rvizconfig" default="$(find wpv3_bringup)/rviz/sensor.rviz" />

  <param name="robot_description" command="$(find xacro)/xacro $(arg model)" />
  <param name="use_gui" value="$(arg gui)"/>

  <node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher" >
    <rosparam command="load" file="$(find wpv3_bringup)/config/wpv3.yaml" />
  </node>
  <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" />
  <node name="rviz" pkg="rviz" type="rviz" args="-d $(arg rvizconfig)" required="true" />

  <!--- Run urg node -->
  <node name="urg_node" pkg="urg_node" type="urg_node" output="screen">
    <param name="ip_address" type="string" value="192.168.0.10"/>
    <remap from="scan" to="scan_raw"/>
  </node>

  <!-- Run lidar filter -->
  <node pkg="wpv3_bringup" type="wpv3_lidar_filter" name="wpv3_lidar_filter">
    <param name="pub_topic" value="/scan"/>
  </node>

</launch>
